crates/primitives/src/transaction/mod.rs

```diff
@@ -1401,6 +1408,115 @@ impl alloy_consensus::Transaction for TransactionSigned {
     }
 }
 
+impl SignedTransaction for TransactionSigned {
+    type Transaction = Transaction;
+    type Signature = Signature;
+
+    fn tx_hash(&self) -> &TxHash {
+        Self::hash_ref(self)
+    }
+
+    fn transaction(&self) -> &Self::Transaction {
+        Self::transaction(self)
+    }
+
+    fn signature(&self) -> &Self::Signature {
+        Self::signature(self)
+    }
+
+    fn recover_signer(&self) -> Option<Address> {
+        Self::recover_signer(self)
+    }
+
+    fn recover_signer_unchecked(&self) -> Option<Address> {
+        Self::recover_signer_unchecked(self)
+    }
+
+    fn from_transaction_and_signature(
+        transaction: Self::Transaction,
+        signature: Self::Signature,
+    ) -> Self {
+        Self::from_transaction_and_signature(transaction, signature)
+    }
+
+    fn fill_tx_env(&self, tx_env: &mut TxEnv, sender: Address) {
```

Comment:
I didn't copy across anything flagged behind `#[cfg(feature = "optimism")]` as I assume that will go into `OpTransactionSigned`